src/types/middleware.ts

```diff
@@ -5,6 +5,7 @@ import type { AtLeastOne } from './utils';
 export type MiddlewareDefinition<TUiState extends UiState = UiState> = {
   onStateChange(options: { uiState: TUiState }): void;
   subscribe(): void;
+  started(): void;
```

Comment:
So this means user-made custom middlewares now need to implement `started()` as well ?
Could it be optional ?

Reply:
It's actually optional in the public definition, but our own middleware use the type without optional keys @aymeric-giraudet

Reply:
see https://github.com/algolia/instantsearch.js/blob/6e2bf7b5cb0f56395938b17d7ef44f986ea45ca1/src/types/middleware.ts#L22 :)